src/Microsoft.Health.Fhir.SqlServer/Features/Schema/SchemaJobWorker.cs

```diff
@@ -0,0 +1,72 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core;
+using Microsoft.Health.Fhir.SqlServer.Configs;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Schema
+{
+    /// <summary>
+    /// The worker responsible for running the schema job.
+    /// </summary>
+    public class SchemaJobWorker
+    {
+        private readonly Func<IScoped<ISchemaDataStore>> _schemaDataStoreFactory;
+        private readonly SqlServerDataStoreConfiguration _sqlServerDataStoreConfiguration;
+        private readonly ILogger _logger;
+
+        public SchemaJobWorker(Func<IScoped<ISchemaDataStore>> schemaDataStoreFactory, SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration, ILogger<SchemaJobWorker> logger)
+        {
+            EnsureArg.IsNotNull(schemaDataStoreFactory, nameof(schemaDataStoreFactory));
+            EnsureArg.IsNotNull(sqlServerDataStoreConfiguration, nameof(sqlServerDataStoreConfiguration));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _schemaDataStoreFactory = schemaDataStoreFactory;
+            _sqlServerDataStoreConfiguration = sqlServerDataStoreConfiguration;
+            _logger = logger;
+        }
+
+        public async Task ExecuteAsync(SchemaInformation schemaInformation, string instanceName, CancellationToken cancellationToken)
+        {
+            using (IScoped<ISchemaDataStore> store = _schemaDataStoreFactory())
+            {
+                await store.Value.InsertInstanceSchemaInformation(instanceName, schemaInformation, cancellationToken);
+            }
+
+            _logger.LogInformation($"Polling started at {Clock.UtcNow}");
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    using (IScoped<ISchemaDataStore> store = _schemaDataStoreFactory())
+                    {
+                        // Ensure schemaInformation has the latest current version
+                        schemaInformation.Current = await store.Value.UpsertInstanceSchemaInformation(instanceName, schemaInformation, cancellationToken);
+
+                        await store.Value.DeleteExpiredRecords();
+                    }
+
+                    await Task.Delay(_sqlServerDataStoreConfiguration.SchemaUpdatesJobPollingFrequency, cancellationToken);
```

Comment:
If an error happens before this, this loop will not wait and will continuously process the error. Thoughts on having the delay outside the using?

Reply:
Ah Good point! I fixed it by adding the delay in finally block to poll every minute even if its error.